src/Aspire.Hosting/Publishing/ManifestPublishingContext.cs

```diff
@@ -269,6 +269,31 @@ public void WritePortBindingEnvironmentVariables(IResource resource)
         }
     }
 
+    internal void WriteDockerBuildArgs(IEnumerable<DockerBuildArg>? buildArgs)
+    {
+        var args = buildArgs?.ToArray() ?? [];
```

Comment:
Just leave it null, the if check on the next line already handles it.
```suggestion
        var args = buildArgs?.ToArray();
```